react-ui/src/components/LogInForm.js

```diff
@@ -34,73 +35,90 @@ class logInForm extends Component {
     });
   };
 
-
   login = event => {
     let email = this.state.loginParams.email;
     let password = this.state.loginParams.password;
-    var params = {email: email, password: password}
-    axios.get(`/public/login/?${queryString.stringify(params)}`)
-    .then(res => {
-      if(res.data!==""){
-        this.setState({
-          email : res.data.email,
-          userType: res.data.userType,
-          firstName: res.data.firstName,
-          personId: res.data.workerId || res.data.studentId,
-          accessToken: res.data.accessToken,
-          islogged: true
+    let validEmail = false;
+
+    if (/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))*$/.test(email)) {
```

Comment:
Hi Kyle,
I would recommend passing the type `email` in the email input field to avoid repeating already existing logic. For an example, please refer to https://www.w3schools.com/tags/att_input_type_email.asp

Reply:
Hi Amy,

I did look at passing the type email, and it valids an email if the user added @ in between the character, so I added that line to cover other extreme cases such as if a user types in asdasd@fsdfdf or joshuabrooks@gmail.c. With only the email type, it wouldn't cover the extreme cases, and as such someone could type in asdasd@fsdfdf and a random password and somehow log in to a blank screen or do joshuabrooks@gmail.c and a password and somehow log in. This would not be ideal, so I keep the logic

Reply:
Hi Kyle,
Makes sense. Awesome, thank you for considering the edge cases!